### PR TITLE
Add ProtonMail config

### PR DESCRIPTION
**Pull request by mkmelin**

### CODE REVIEW

**Comment by benbucksch** [comment]
I'm not sure how useful it is to add a config that works only if you install some other Windows software on your computer.

We can add it once protonmail has a proper IMAP server that is reachable over the Internet.

**Comment by benbucksch** [comment]
FYI, https://proton.me/mail/bridge states: "Proton Mail Bridge is currently available for paid subscribers." What happens, if I download one of those [Open Source builds](https://github.com/ProtonMail/proton-bridge/releases) above and I'm not a paid subscriber?

**Comment by Sancus** [comment]
This should be renamed with a .xml extension as the bot says, too.

**Comment by mkmelin** [comment]
> FWIW, this config works only in combination with a third party Windows/Linux/Mac application called [Proton Mail Bridge](https://proton.me/mail/bridge), which has to be installed separately, before this config can work.

Correct.
If you don't have the bridge running setting up the account won't work. 
Adding this config can't change that. But it means all you have to do is start the bridge and then set up as normal without looking for hostnames ports etc on the website.

**Comment by benbucksch** [comment]
Given that you have to install a native application *first* anyway, the installer of this Proton Bridge could find the Thunderbird install directory, and drop this file in Thunderbird install dir `/isp/protonmail.com.xml` . This way, the config is presented only to those users who can actually use it.

File: .github/workflows/validate-config.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+  validate_config:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Validate file extensions
+      run: | 
+       set -o pipefail
```
Comment (benbucksch) [inline .github/workflows/validate-config.yml:14]: This file doesn't belong in this PR

File: ispdb/protonmail.com.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+       <hostname>127.0.0.1</hostname>
+       <port>1025</port>
+       <socketType>STARTTLS</socketType>
+       <username>%EMAILADDRESS%</username>
+    </outgoingServer>
+
+    <enable visiturl="https://protonmail.com/bridge/">
+      <instruction>You must first install and activate the ProtonMail Bridge.</instruction>
+    </enable>
+
```
Comment (benbucksch) [inline ispdb/protonmail.com.xml:28]: Alternatively, Thunderbird account config UI could implement showing this `<enable>` message, with a button to open the URL in a browser. That would lead the users the right way.

Thank you for adding this information to the config!

**Comment by benbucksch** [comment]
Alternatively, Thunderbird account config UI could implement showing this <enable> message, with a button to open the URL in a browser. That would lead the users the right way.

**Comment by benbucksch** [comment]
I've posted a proposal to add the enable message to the Thunderbird account setup UI.

File: .github/workflows/validate-config.yml (hunk 1 above)
Comment (mkmelin) [reply .github/workflows/validate-config.yml:14]: Ah, rebase error.

**Comment by mkmelin** [comment]
> Given that you have to install a native application _first_ anyway, the installer of this Proton Bridge could find the Thunderbird install directory, and drop this file in Thunderbird install dir `/isp/protonmail.com.xml` . This way, the config is presented only to those users who can actually use it.

It could. But it's not.

**Comment by Sancus** [comment]
The current situation is that you don't get a working config when you enter your protonmail address. So I don't see much problem with including this, as it can only make things better.

**Comment by cketti** [comment]
When visiting the [Protonmail documentation for Thunderbird](https://protonmail.com/bridge/thunderbird) just now, I saw the following screenshot:

image

And indeed, they're now hosting their config under…
- https://autoconfig.protonmail.com/mail/config-v1.1.xml
- https://autoconfig.protonmail.ch/mail/config-v1.1.xml
- https://autoconfig.pm.me/mail/config-v1.1.xml

I think with that we can close this pull request. Does anyone disagree?

**Comment by Sancus** [comment]
> I think with that we can close this pull request. Does anyone disagree?

Seems reasonable.

**Comment by benbucksch** [comment]
Agreed
